### PR TITLE
Changed docker-machine and sshutil to support hostnames with colons i…

### PR DESCRIPTION
**Pull request by aaron-prindle**
…n them, for example ipv6 addresses

### CODE REVIEW

File: pkg/minikube/sshutil/sshutil.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func NewSSHClient(d drivers.Driver) (*ssh.Client, error) {
 		return nil, errors.Wrapf(err, "Error creating new native config from ssh using: %s, %s", h.Username, auth)
 	}
 
-	client, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", h.IP, h.Port), &config)
+	client, err := ssh.Dial("tcp", fmt.Sprintf("[%s]:%d", h.IP, h.Port), &config)
```
Comment (jimmidyson) [inline pkg/minikube/sshutil/sshutil.go]: I think you're better off using `net.JoinHostPort` here.

File: vendor/github.com/docker/machine/libmachine/ssh/client.go [hunk 2]
```diff
@@ -166,7 +166,7 @@ func (client *NativeClient) session(command string) (*ssh.Session, error) {
 		return nil, fmt.Errorf("Error attempting SSH client dial: %s", err)
 	}
 
-	conn, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", client.Hostname, client.Port), &client.Config)
+	conn, err := ssh.Dial("tcp", fmt.Sprintf("[%s]:%d", client.Hostname, client.Port), &client.Config)
```
Comment (jimmidyson) [inline vendor/github.com/docker/machine/libmachine/ssh/client.go]: Use `net.JoinHostPort`.

File: vendor/github.com/docker/machine/libmachine/ssh/client.go [hunk 3]
```diff
@@ -250,7 +250,7 @@ func (client *NativeClient) Shell(args ...string) error {
 	var (
 		termWidth, termHeight int
 	)
-	conn, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", client.Hostname, client.Port), &client.Config)
+	conn, err := ssh.Dial("tcp", fmt.Sprintf("[%s]:%d", client.Hostname, client.Port), &client.Config)
```
Comment (jimmidyson) [inline vendor/github.com/docker/machine/libmachine/ssh/client.go]: Use `net.JoinHostPort`.

**Comment by aaron-prindle** [comment]
Now using net.JoinHostPort.  Thanks!

**Comment by aaron-prindle** [comment]
https://github.com/docker/machine/pull/3783

**Pull request merged**